From fecad2b4a23d62db51986b772cd8222c141b9c65 Mon Sep 17 00:00:00 2001
From: Kamlakant Patel <kamlakant.patel@broadcom.com>
Date: Tue, 9 Dec 2014 20:22:57 +0530
Subject: spi: mnxlp8xx: Enable SPI controller

Add node for SPI and its slave devices SPI NOR and SPI NAND.
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Nam Ninh <nam.ninh@windriver.com>

diff --git a/arch/mips/boot/dts/xlp8xx-linux-mn-kernel.dts b/arch/mips/boot/dts/xlp8xx-linux-mn-kernel.dts
index 2561e78..788cd83 100644
--- a/arch/mips/boot/dts/xlp8xx-linux-mn-kernel.dts
+++ b/arch/mips/boot/dts/xlp8xx-linux-mn-kernel.dts
@@ -173,6 +173,65 @@
 			};
 		};
 
+		spi: xlp_spi@3a000 {
+			compatible = "netlogic,xlp-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0 0x3a000 0x100>;
+
+			interrupts = <34>;
+			interrupt-parent = <&pic0>;
+
+			spi_nor@1 {
+				compatible = "spansion,s25sl12801";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				reg = <1>;		/* Chip Select */
+				spi-max-frequency = <40000000>;
+
+				partition@0 {
+					label = "x-loader";
+					reg = <0x0 0x200000>; /* 2M */
+					read-only;
+				};
+
+				partition@200000 {
+					label = "kernel";
+					reg = <0x200000 0x500000>; /* 5M */
+				};
+
+				partition@700000 {
+					label = "rootfs";
+					reg = <0x700000 0x800000>; /* 8M */
+				};
+
+				partition@f00000 {
+					label = "env";
+					reg = <0xf00000 0x100000>; /* 1M */
+					read-only;
+				};
+			};
+
+			spi_nand@2 {
+				compatible = "spinand,mt29f";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				reg = <2>;	/* SPI NAND CS */
+				spi-max-frequency = <50000000>;
+
+				partition@0 {
+					label = "firmware";
+					reg = <0x0 0x1c00000>;
+				};
+
+				partition@1 {
+					label = "kernel";
+					reg = <0x1c00000 0x0>;
+				};
+			};
+		};
+
 		gpio0: xlp_gpio@34000 {
 			compatible = "netlogic,xlp-gpio";
 			reg = <0 0x34100 0x1000>;
diff --git a/arch/mips/boot/dts/xlp8xx-linux-mn4-kernel.dts b/arch/mips/boot/dts/xlp8xx-linux-mn4-kernel.dts
index 955db65..6f9e6b3 100644
--- a/arch/mips/boot/dts/xlp8xx-linux-mn4-kernel.dts
+++ b/arch/mips/boot/dts/xlp8xx-linux-mn4-kernel.dts
@@ -173,6 +173,65 @@
 			};
 		};
 
+		spi: xlp_spi@3a000 {
+			compatible = "netlogic,xlp-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0 0x3a000 0x100>;
+
+			interrupts = <34>;
+			interrupt-parent = <&pic0>;
+
+			spi_nor@1 {
+				compatible = "spansion,s25sl12801";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				reg = <1>;		/* Chip Select */
+				spi-max-frequency = <40000000>;
+
+				partition@0 {
+					label = "x-loader";
+					reg = <0x0 0x200000>; /* 2M */
+					read-only;
+				};
+
+				partition@200000 {
+					label = "kernel";
+					reg = <0x200000 0x500000>; /* 5M */
+				};
+
+				partition@700000 {
+					label = "rootfs";
+					reg = <0x700000 0x800000>; /* 8M */
+				};
+
+				partition@f00000 {
+					label = "env";
+					reg = <0xf00000 0x100000>; /* 1M */
+					read-only;
+				};
+			};
+
+			spi_nand@2 {
+				compatible = "spinand,mt29f";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				reg = <2>;	/* SPI NAND CS */
+				spi-max-frequency = <50000000>;
+
+				partition@0 {
+					label = "firmware";
+					reg = <0x0 0x1c00000>;
+				};
+
+				partition@1 {
+					label = "kernel";
+					reg = <0x1c00000 0x0>;
+				};
+			};
+		};
+
 		gpio0: xlp_gpio@34000 {
 			compatible = "netlogic,xlp-gpio";
 			reg = <0 0x34100 0x1000>;
-- 
1.7.1

